userspace/libscap/ringbuffer/devset.c

```diff
@@ -28,11 +21,11 @@ int32_t devset_init(struct scap_device_set *devset, size_t num_devs, char *laste
 
 	for(size_t j = 0; j < num_devs; ++j)
 	{
-		devset->m_devs[j].m_buffer = MAP_FAILED;
-		devset->m_devs[j].m_bufinfo = MAP_FAILED;
-		devset->m_devs[j].m_bufstatus = MAP_FAILED;
-		devset->m_devs[j].m_fd = -1;
-		devset->m_devs[j].m_bufinfo_fd = -1;
+		devset->m_devs[j].m_buffer = INVALID_MAPPING;
+		devset->m_devs[j].m_bufinfo = INVALID_MAPPING;
+		devset->m_devs[j].m_bufstatus = INVALID_MAPPING;
+		devset->m_devs[j].m_fd = INVALID_FD;
+		devset->m_devs[j].m_bufinfo_fd = INVALID_FD;
```

Comment:
I don't understand why some of these refactorings of devset handling are part of this PR 🤔. Don't get me wrong, they are awesome and greatly simplify the code, I just wish they were in their own PR so they could get a bit more of the spotlight.

Reply:
Yeah, that's what my last comment is about :) I can submit it separately after just the scap_strerror cleanup if you prefer (I'd just rather not do another series of rebase dances and it depends a bit on the scap_strerror cleanup).

It does fix rather important Windows warnings (how in the world didn't we get a link error?) so I'd rather see it in sooner than later but I'm fine with a separate PR.